backend/app/services/posting_service.rb

```diff
@@ -65,14 +65,71 @@ def prefill(user:)
         applicant = Applicant.find_by(utorid: utorid)
         return { utorid: utorid } unless applicant
 
-        {
+        # This is the basic data we can get if the applicant already exists
+        data = {
             utorid: utorid,
             student_number: applicant.student_number,
             first_name: applicant.first_name,
             last_name: applicant.last_name,
             email: applicant.email,
             phone: applicant.phone
         }
+
+        existing_application = Application.find_by(posting: @posting)
+        if existing_application
+            application_service =
+                ApplicationService.new application: existing_application
+            data.merge! application_service.prefilled_data
+        else
+            # Find out if they have previous contracts in the database.
+            # This will tell us information about whether the applicant has been previously employed
+            offer_history =
+                Assignment.joins(:applicant, :offers, :position).where(
+                    applicant: applicant, "offers.status": 'accepted'
+                ).order("offers.position_start_date": :ASC).pluck(
```

Comment:
Pretty cool ORM stuff! Do I understand correctly that `pluck` is like SELECT for specific columns?

Reply:
Yes. That's exactly what it does. You can try it out in the rails console and see the query that it produces :-)